lib/components/user/saved-trip-list.js

```diff
@@ -1,72 +1,130 @@
+ * This class manages events and rendering for one item in the saved trip list.
  */
-class SavedTripList extends Component {
+class TripListItem extends Component {
   /**
    * Navigate to the saved trip's URL #/savedtrips/trip-id-123.
    * (There shouldn't be a need to encode the ids from Mongo.)
    */
-  _handleTripSelect = trip => () => {
-    const { id } = trip
-    this.props.routeTo(`/savedtrips/${id}`)
+  _handleEditTrip = () => {
+    const { routeTo, trip } = this.props
+    routeTo(`/savedtrips/${trip.id}`)
   }
 
-  render () {
-    const { trips } = this.props
-
-    // TODO: Improve navigation.
-    const accountLink = <p><LinkButton to='/account'>Back to My Account</LinkButton></p>
-    let content
-
-    if (!trips || trips.length === 0) {
-      content = (
-        <>
-          {accountLink}
-          <h1>You have no saved trips</h1>
-          <p>Perform a trip search from the map first.</p>
-        </>
-      )
-    } else {
-      // Stack the saved trip summaries. When the user clicks on one, they can edit that trip.
-      content = (
-        <>
-          {accountLink}
-          <h1>My saved trips</h1>
-          <p>Click on a saved trip below to modify it.</p>
-          <ButtonGroup vertical block>
-            {trips.map((trip, index) => (
-              <Button key={index} onClick={this._handleTripSelect(trip)} style={{textAlign: 'left'}}>
-                <TripSummaryPane monitoredTrip={trip} />
-              </Button>
-            ))}
-          </ButtonGroup>
-        </>
-      )
+  /**
+   * Pauses or resumes the specified trip.
+   */
+  _handlePauseOrResumeMonitoring = () => {
+    const { createOrUpdateUserMonitoredTrip, trip } = this.props
+    const newTrip = clone(trip)
+    newTrip.isActive = !newTrip.isActive
+
+    // Silent update of existing trip.
+    createOrUpdateUserMonitoredTrip(newTrip, false, true)
+  }
+
+  /**
+   * Deletes a trip from persistence.
+   * (The operation also refetches the redux monitoredTrips for the logged-in user.)
+   */
+  _handleDeleteTrip = async () => {
+    if (confirm('Would you like to remove this trip?')) {
+      const { deleteUserMonitoredTrip, trip } = this.props
+      await deleteUserMonitoredTrip(trip.id)
     }
+  }
 
+  render () {
+    const { trip } = this.props
     return (
-      <div className='otp'>
-        {/* TODO: Do mobile view. */}
-        <DesktopNav />
-        <form className='container'>
-          {content}
-        </form>
-      </div>
+      <Panel>
+        <Panel.Heading>
+          <Panel.Title componentClass='h3'>{trip.tripName}</Panel.Title>
+        </Panel.Heading>
+        <Panel.Body>
+          <TripSummaryPane monitoredTrip={trip} />
+          <ButtonGroup>
+            <Button bsSize='small' onClick={this._handlePauseOrResumeMonitoring}>
+              {trip.isActive
+                ? <><Glyphicon glyph='pause' /> Pause notifications</>
+                : <><Glyphicon glyph='play' /> Resume notifications</>
```

Comment:
Remove "notifications" from each.

Reply:
If the buttons just say "Pause" or "Resume", could someone read the buttons as "Pause trip" or "Resume trip"?

Reply:
Maybe with this layout it's ok as the buttons are just underneath the "Notifications" label.